src/Microsoft.Graph.Core/Extensions/BaseRequestExtensions.cs

```diff
@@ -0,0 +1,142 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using System;
+    using System.Net.Http;
+
+    /// <summary>
+    /// Extension methods for <see cref="BaseRequest"/>
+    /// </summary>
+    public static class BaseRequestExtensions
+    {
+        /// <summary>
```

Comment:
Are there any assumptions that the dev should know about before they can use WithScopes? For example, does it assume they are using the default auth middleware, or that they have implemented anything beforehand? If so, please specify with the assumptions are for using this

Reply:
@MIchaelMainer I have updated the summaries to emphasize on their usage.